### PR TITLE
add faq for gcp authorization

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

The `AdmissionReview` seems to not work for the default administrator for gcp cluster. A Q&A is added in this PR. Besides, the faq in `1.2.3` is also copied here ( I don't know whether it should be ).

Related issue: https://github.com/chaos-mesh/chaos-mesh/issues/2173 , https://github.com/chaos-mesh/chaos-mesh/issues/2187

### CODE REVIEW

File: versioned_docs/version-2.0.1/faqs.md [hunk 1]
```diff
@@ -1,6 +1,111 @@
 ---
-title: 常见问题解答 (FAQ)
-sidebar_label: 常见问题解答 (FAQ)
+id: faqs
+title: FAQs
+sidebar_label: FAQs
 ---
 
-## TODO
+## Question
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:7]:
```suggestion
## Questions
```

File: versioned_docs/version-2.0.1/faqs.md [hunk 2]
```diff
@@ -1,6 +1,111 @@
...
+
+### Q: If I do not have Kubernetes clusters deployed, can I use Chaos Mesh to create chaos experiments?
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:9]:
```suggestion
### Q: If I do not have deployed Kubernetes clusters, can I use Chaos Mesh to create chaos experiments?
```

File: versioned_docs/version-2.0.1/faqs.md [hunk 3]
```diff
@@ -1,6 +1,111 @@
...
+
+No, you can not use Chaos Mesh in this case. But still you can run chaos experiments using command line. Refer to [Command Line Usages of Chaos](https://github.com/pingcap/tipocket/blob/master/doc/command_line_chaos.md) for details.
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:11]:
```suggestion
No, you cannot use Chaos Mesh in this case. But you still can run chaos experiments using command lines. For more details, see [Command Line Usages of Chaos](https://github.com/pingcap/tipocket/blob/master/doc/command_line_chaos.md).
```

File: versioned_docs/version-2.0.1/faqs.md [hunk 4]
```diff
@@ -1,6 +1,111 @@
...
+
+### Q: I have deployed Chaos Mesh and created PodChaos experiments successfully, but I still failed in creating NetworkChaos/TimeChaos Experiment. The log is shown below:
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:13]:
```suggestion
### Q: I have deployed Chaos Mesh and created PodChaos experiments successfully, but I still failed in creating NetworkChaos/TimeChaos Experiment. The log is shown as below:
```

File: versioned_docs/version-2.0.1/faqs.md [hunk 5]
```diff
@@ -1,6 +1,111 @@
...
+
+```
+2020-06-18T01:05:26.207Z	ERROR	controllers.TimeChaos	failed to apply chaos on all pods	{"reconciler": "timechaos", "error": "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp xx.xx.xx.xx:xxxxx: connect: connection refused\""}
+```
+
+You can try using the parameter: `hostNetwork`, as shown below:
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:19]:
```suggestion
You can try using the `hostNetwork` parameter as following:
```

File: versioned_docs/version-2.0.1/faqs.md [hunk 6]
```diff
@@ -1,6 +1,111 @@
...
+
+```
+# vim helm/chaos-mesh/values.yaml, change hostNetwork from false to true
+hostNetwork: true
+```
+
+### Q: I just saw `ERROR: failed to get cluster internal kubeconfig: command "docker exec --privileged kind-control-plane cat /etc/kubernetes/admin.conf" failed with error: exit status 1` when installing Chaos Mesh with kind. How to fix it?
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:26]:
```suggestion
### Q: I just saw `ERROR: failed to get cluster internal kubeconfig: command "docker exec --privileged kind-control-plane cat /etc/kubernetes/admin.conf" failed with error: exit status 1` when installing Chaos Mesh with kind. How should I  fix it?
```

File: versioned_docs/version-2.0.1/faqs.md [hunk 7]
```diff
@@ -1,6 +1,111 @@
...
+
+You can try the following command to fix it:
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:28]:
```suggestion
You can use the following command to fix it:
```

File: versioned_docs/version-2.0.1/faqs.md [hunk 8]
```diff
@@ -1,6 +1,101 @@
...
+You can use the `hostNetwork` parameter to fix this issue as follows:
+
+```
+# vim helm/chaos-mesh/values.yaml, change hostNetwork from false to true
+hostNetwork: true
+```
+
+### Q: The default administrator Google Cloud user account is forbidden to create chaos experiments. How to fix it?
+
+The default administrator Google Cloud user cannot be checked by `AdmissionReview`. So you have to create a administrator role and bind it to the account to grant it the priviledge to create chaos experiments. For example:
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:28]:
```suggestion
The default administrator Google Cloud user cannot be checked by `AdmissionReview`. You need to create an administrator role and assign the role to the account to grant the privilege of creating chaos experiments to it. For example:
```

File: versioned_docs/version-2.0.1/faqs.md [hunk 9]
```diff
@@ -1,6 +1,101 @@
...
+  kind: ClusterRole
+  name: role-cluster-manager-pdmas
+  apiGroup: rbac.authorization.k8s.io
+```
+
+The `USER_ACCOUNT` above should be your Google Cloud user email.
+
+## DNSChaos
+
+### Q: While trying to run DNSChaos in OpenShift, tripped over problems regarding authorization.
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:63]:
```suggestion
### Q: While trying to run DNSChaos in OpenShift, the problems regarding authorization blocked the process.
```
这部分修改为这样会不会影响原语意呢？

File: versioned_docs/version-2.0.1/faqs.md [hunk 10]
```diff
@@ -1,6 +1,101 @@
...
+
+Message most looked like this:
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:65]:
```suggestion
The error message is similar to this:
```

File: versioned_docs/version-2.0.1/faqs.md [hunk 11]
```diff
@@ -1,6 +1,101 @@
...
+Error creating: pods "chaos-dns-server-123aa56123-" is forbidden: unable to validate against any security context constraint: [spec.containers[0].securityContext.capabilities.add: Invalid value: "NET_BIND_SERVICE": capability may not be added]
+```
+
+You need to add privileged scc to default.
+
+```bash
+oc adm policy add-scc-to-user privileged -n chaos-testing -z chaos-dns-server
+```
+
+## Install
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:77]:
```suggestion
## Installation
```

File: versioned_docs/version-2.0.1/faqs.md [hunk 12]
```diff
@@ -1,6 +1,101 @@
...
+
+### Q: While trying to install chaos-mesh in OpenShift, tripped over problems regarding authorization.
+
+Message most looked like this:
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:81]:
```suggestion
The error message is similar to this:
```

File: versioned_docs/version-2.0.1/faqs.md [hunk 13]
```diff
@@ -1,6 +1,101 @@
...
+
+You need to add privileged scc to default.
+
+```bash
+oc adm policy add-scc-to-user privileged -n chaos-testing -z chaos-dns-server
+```
+
+## Install
+
+### Q: While trying to install chaos-mesh in OpenShift, tripped over problems regarding authorization.
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:79]:
```suggestion
### Q: While trying to install Chaos Mesh in OpenShift, the problems regarding authorization blocked the installation process.
```

File: versioned_docs/version-2.0.1/faqs.md [hunk 14]
```diff
@@ -1,6 +1,101 @@
...
+title: FAQs
+sidebar_label: FAQs
 ---
 
-## TODO
+## Questions
+
+### Q: If I do not have deployed Kubernetes clusters, can I use Chaos Mesh to create chaos experiments?
+
+No, you cannot use Chaos Mesh in this case. But you still can run chaos experiments using command lines. For more details, see [Command Line Usages of Chaos](https://github.com/pingcap/tipocket/blob/master/doc/command_line_chaos.md).
```
Comment (WangXiangUSTC) [inline versioned_docs/version-2.0.1/faqs.md:11]: 这里能否也加一下 chaosd 链接

File: versioned_docs/version-2.0.1/faqs.md [hunk 15]
```diff
@@ -1,6 +1,101 @@
...
+### Q: I have deployed Chaos Mesh and created PodChaos experiments successfully, but I still failed in creating NetworkChaos/TimeChaos Experiment. The log is shown as below:
+
+```
+2020-06-18T01:05:26.207Z	ERROR	controllers.TimeChaos	failed to apply chaos on all pods	{"reconciler": "timechaos", "error": "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp xx.xx.xx.xx:xxxxx: connect: connection refused\""}
+```
+
+You can use the `hostNetwork` parameter to fix this issue as follows:
+
+```
+# vim helm/chaos-mesh/values.yaml, change hostNetwork from false to true
```
Comment (WangXiangUSTC) [inline versioned_docs/version-2.0.1/faqs.md:22]: 是否需要加一个 helm upgrade 的说明，因为有些用户安装时没有clone 仓库

**Review by WangXiangUSTC — COMMENTED** [review]
rest LGTM

File: versioned_docs/version-2.0.1/faqs.md [hunk 16]
```diff
@@ -1,6 +1,101 @@
...
+
+### Q: While trying to run DNSChaos in OpenShift, tripped over problems regarding authorization.
+
+Message most looked like this:
+
+```bash
+Error creating: pods "chaos-dns-server-123aa56123-" is forbidden: unable to validate against any security context constraint: [spec.containers[0].securityContext.capabilities.add: Invalid value: "NET_BIND_SERVICE": capability may not be added]
+```
+
+You need to add privileged scc to default.
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:71]:
```suggestion
In this case, you need to add privileged Security Context Constraints (SCC) to the `chaos-dns-server`.
```

File: versioned_docs/version-2.0.1/faqs.md [hunk 17]
```diff
@@ -1,6 +1,101 @@
...
+ Invalid value: true: Host PID is not allowed to be used spec.securityContext.hostIPC:
+ Invalid value: true: Host IPC is not allowed to be used securityContext.runAsUser:
+ Invalid value: "hostPath": hostPath volumes are not allowed to be used spec.containers[0].securityContext.volumes[1]:
+ Invalid value: true: Host network is not allowed to be used spec.containers[0].securityContext.containers[0].hostPort:
+ Invalid value: 31767: Host ports are not allowed to be used spec.containers[0].securityContext.hostPID:
+ Invalid value: true: Host PID is not allowed to be used spec.containers[0].securityContext.hostIPC:
+......]
+```
+
+You need to add privileged scc to default.
```
Comment (en-jin19) [inline versioned_docs/version-2.0.1/faqs.md:96]:
```suggestion
you need to add the privileged Security Context Constraints (SCC) to the `chaos-daemon`
```

File: versioned_docs/version-2.0.1/faqs.md (hunk 15 above)
Comment (YangKeao) [reply versioned_docs/version-2.0.1/faqs.md]: 对

File: versioned_docs/version-2.0.1/faqs.md (hunk 14 above)
Comment (YangKeao) [reply versioned_docs/version-2.0.1/faqs.md]: 俺把这句话删了，直接改成 chaosd 吧

**Comment by YangKeao** [comment]
@WangXiangUSTC PTAL

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

**Review by en-jin19 — APPROVED** [review]
LGTM

**Pull request merged**